### PR TITLE
73848 Checkbox and loop for Gulf War 1990 location + dates pages

### LINKED ISSUE #73848: Toxic Exposure: Gulf War Hazard 1990 Date Ranges
## Issue Description
- Implement checkbox and loop flow for the Gulf War Hazard 1990 question.
- For each location selected (except for "None of these locations" and "I'm not sure"), display a follow up page to ask for the approximate date range served in that location.
- None and I'm not sure checkboxes are out of scope for this story. Will be added in subsequent story/ies.
- Discovery on what type of automated tests we can add for this. Create follow up story if needed.

## Front end tasks
- Ensure the page is styled according to design spec. 
- ~Ensure the back end data from API is displayed in local and review instance~
- Ensure unit tests are available. 
- Ensure your code changes are covered by E2E tests
- Run axe checks using the Chrome or Firefox browser plugin
- Test for color contrast and color blindness issues
- Zoom layouts to 400% at 1280px width
- Test with 1 or 2 screen readers
- Navigate using the keyboard only

## Acceptance Criteria
- [ ] Designer comments on this issue to approve page styling 
- [ ] Unit tests pass
- [ ] E2E tests cover current code and regression
- [ ] End-to-end tests show 0 violations.
- [ ] ~The data returned matches API response (for given user or scenario)~
- [ ] All axe checks pass
- [ ] All color contrast checks pass
- [ ] All zoom testing passes
- [ ] All keyboard checks pass 
- [ ] All screenreader checks pass

## Appendix
<details> 

 <summary>Ensure your code changes are covered by E2E tests (expand)</summary>
 
 - Add E2E tests if none exist for this addition/change. 
 
 - Update existing E2E tests if this code will change functionality. 
 
 - Include axe checks, including hidden content
 
</details>

<details> 

 <summary> Run axe checks using the Chrome or Firefox browser plugin (expand)</summary>
 
- Ensure no heading levels are skipped.

- Ensure all buttons and labeled inputs use semantic HTML elements.

- Ensure all buttons, labeled elements and images are identified using HTML semantic markers or ARIA roles.

- Ensure form fields have clearly defined boundaries or outlines.

- Ensure form fields do not use placeholder text. 

- Ensure form fields have highly visible and specific error states. 

</details>

<details> 

 <summary> Test for color contrast and color blindness issues (expand) </summary>
 - All text has appropriate contrast. 

</details>

<details> 

 <summary> Zoom layouts to 400% at 1280px width (expand)</summary> 
 - Ensure readability and usability are supported when zoomed up to 400% at 1280px browser width
 - Ensure no content gets focused offscreen or is hidden from view.
 
</details>

<details> 

 <summary> Test with 1 or 2 screen readers (expand)</summary> 
 
 - Ensure the page includes a skip navigation link. 
 
 - Ensure all links are properly descriptive. 
 
 - Ensure screen reader users can hear the text equivalent for each image conveying information. 

 - Ensure screen reader users can hear the text equivalent for each image button. 

 - Ensure screen reader users can hear labels and instructions for inputs. 
 
 - Ensure purely decorative images are not announced by the screenreader. 
 
</details>

<details> 

 <summary>Navigate using the keyboard only (expand)</summary> 
 
 - Ensure all links (navigation, text and/or image), form controls and page functions can be reached with the tab key in a logical order. 
 
 - Ensure all links (navigation, text and/or image), form controls and page functions can be triggered with the spacebar, enter key, or arrow keys.
 
 - Ensure all interactive elements can be reached with the tab key in a logical order 
 
 - Ensure all interactive elements can be triggered with the spacebar, enter key, or arrow keys.
 
 - Ensure focus is always visible and appears in logical order.
 
 - Ensure each interactive element has visible focus state which appears in logical order.
 

</details>

---
## How to configure this issue
- [ ] **Attached to a Milestone** (when will this be completed?)
- [ ] **Attached to an Epic** (what body of work is this a part of?)
- [ ] **Labeled with Team** (`product support`, `analytics-insights`, `operations`, `service-design`, `Console-Services`, `tools-fe`)
- [ ] **Labeled with Practice Area** (`backend`, `frontend`, `devops`, `design`, `research`, `product`, `ia`, `qa`, `analytics`, `contact center`, `research`, `accessibility`, `content`)
- [ ] **Labeled with Type** (`bug`, `request`, `discovery`, `documentation`, etc.)

### PR DESCRIPTION
**Pull request by christinec-fftc**
## Summary
- Adds a new page for each of the 17 Gulf War 1990 locations. Each page only displays if the Veteran selects the checkbox for that location on the previous page.
- Reference POC https://github.com/department-of-veterans-affairs/vets-website/pull/27609
- This is temporarily using the va-memorable-date component since there currently is no web component pattern for va-date
- _(Which team do you work for, does your team own the maintenance of this component?)_
@department-of-veterans-affairs/dbex-trex 
- _(If using a flipper, what is the end date of the flipper being required/success criteria being targeted)_
toggle cleanup department-of-veterans-affairs/va.gov-team#65089

## Related issue(s)

- _Link to ticket created in va.gov-team repo_
department-of-veterans-affairs/va.gov-team#73848

## Testing done
- _Describe the steps required to verify your changes are working as expected_
Unit tests added. Manual and a11y tests performed
- _Describe the tests completed and the results_

**TESTING FOR NEW FEATURES**
**Setup steps**
1. Enable toggle
2. Start a new claim with new conditions
3. On the `toxic-exposure-conditions` page, select at least one new condition

**No locations selected**
1. On the `gulf-war-hazard-1990` page, do not select any locations OR click 'Continue' to skip the question
2. verify no location dates pages display

**Location(s) selected**
1. On the `gulf-war-hazard-1990` page, select one or more locations
2. Verify a location dates page displays for each location
3. Verify the subtitle displays appropriate counts and location, based on your selection, e.g. '1 of 3: Egypt'

**No dates**
1. Do not enter any dates and click 'Continue'
2. Verify you are able to go to the next page in the flow (i.e. the date fields are optional) and you can continue on with the rest of the form beyond the TE pages

**Both dates or one date**
1. Enter a date for one or both date fields
2. Verify you are able to go to the next page in the flow

**Review and submit**
-Perform similar steps and verification on the review and submit page for these new pages

**REGRESSION**
**toggle DISABLED**
start a new claim
1. TE pages do not display for any claim type 
2. When looking at redux object, `toxicExposureConditions` and `gulfWar1990*` objects are empty 

**toggle ENABLED**
1. claim type new 
TE pages displayed 

2. claim type CFI
no TE pages displayed 

3. both claim types
TE pages displayed 

## Screenshots
screenshot1
image

## What areas of the site does it impact?
526EZ

### Quality Assurance & Testing

- [ ] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:

- [ ] The vets-website header does not contain any web-components
- [ ] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
- [ ] I reached out in the `#sitewide-public-websites` Slack channel for questions

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: src/applications/disability-benefits/all-claims/pages/toxicExposure/gulfWar1990LocationDates.js [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+          uiSchema: makeUiSchema(locationId),
+          schema: makeSchema(locationId),
+          depends: formData =>
+            showGulfWar1990LocationDatesPage(formData, locationId),
+        },
+      };
+    },
+  );
+
+  return Object.assign({}, ...gulfWar1990LocationPagesList);
```
Comment (Mottie) [inline src/applications/disability-benefits/all-claims/pages/toxicExposure/gulfWar1990LocationDates.js:124]: I don't think this `Object.assign` is necessary. You're already building the list from scratch, or am I missing something?

File: src/applications/disability-benefits/all-claims/tests/content/toxicExposure.unit.spec.jsx [hunk 2]
```diff
@@ -180,7 +200,7 @@ describe('toxicExposure', () => {
         },
       };
 
-      window.sessionStorage.setItem(SHOW_TOXIC_EXPOSURE, true);
+      window.sessionStorage.setItem(SHOW_TOXIC_EXPOSURE, 'true');
```
Comment (Mottie) [inline src/applications/disability-benefits/all-claims/tests/content/toxicExposure.unit.spec.jsx:203]: MIght be cleaner to add this in the `beforeEach` callback?

File: src/applications/disability-benefits/all-claims/tests/content/toxicExposure.unit.spec.jsx [hunk 3]
```diff
@@ -317,4 +337,173 @@ describe('toxicExposure', () => {
...
+
+  describe('dateRangePageDescription', () => {
+    it('displays description when counts specified', () => {
+      const tree = render(dateRangePageDescription(1, 5, 'Egypt'));
+      tree.getByText('1 of 5: Egypt', { exact: false });
+      tree.getByText(dateHelp);
+    });
+
+    it('displays description when counts not specified', () => {
+      const tree = render(dateRangePageDescription(0, -1, 'Egypt'));
```
Comment (Mottie) [inline src/applications/disability-benefits/all-claims/tests/content/toxicExposure.unit.spec.jsx:349]: I don't think the `totalItems` value will ever be negative? And as I commented within that function, we're gonna see `1 of 1: Egypt` in the case where `dateRangePageDescription(1, 1, 'Egypt')` is used

**Review by Mottie — DISMISSED** [review]
Looks good. I think some E2E tests should be included in this PR

File: src/applications/disability-benefits/all-claims/content/toxicExposure.jsx [hunk 4]
```diff
@@ -58,10 +30,94 @@ export const conditionsDescription = (
...
+  currentItem,
+  totalItems,
+  locationName,
+) {
+  return (
+    <>
+      <h4 className="vads-u-font-size--h5 vads-u-margin-top--2">
+        {currentItem > 0 &&
+          totalItems > 0 &&
+          `${currentItem} of ${totalItems}: `}
```
[the remark this answers is not in the document]
Comment (christinec-fftc) [reply src/applications/disability-benefits/all-claims/content/toxicExposure.jsx:74]: Great question. I will check with design & product.

File: src/applications/disability-benefits/all-claims/tests/content/toxicExposure.unit.spec.jsx (hunk 2 above)
Comment (christinec-fftc) [reply src/applications/disability-benefits/all-claims/tests/content/toxicExposure.unit.spec.jsx:203]: for sure. i can consolidate the enabled tests into their own section

**Comment by christinec-fftc** [comment]
> Looks good. I think some E2E tests should be included in this PR

As we discussed, there's some initial setup work to do. I brought this up with the team and we refined this ticket [#67019](https://github.com/department-of-veterans-affairs/va.gov-team/issues/67019)

**Review by sethdarragile6 — DISMISSED** [review]
Pulled down and tested locally. Looks great, nice work!

File: src/applications/disability-benefits/all-claims/content/toxicExposure.jsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (christinec-fftc) [reply src/applications/disability-benefits/all-claims/content/toxicExposure.jsx:74]: @Mottie Discussed in our review meeting today and they said they are ok with 1 of 1.  And actually the post 9/11 mocks have this as an example. So will leave as is for now. Thanks!

**Review by rhasselle-oddball — DISMISSED** [review]
looks good!

**Comment by christinec-fftc** [comment]
I happened to do a `yarn install` today and noticed there's an new, odd border around the checkbox groups. Perhaps came with the recent component library version bump. Honoring the `uswds` flag at the checkbox group level seems to fix this, so adding to this pr. fyi @rhasselle-oddball 
image

**Review by powellkerry — APPROVED** [review]

**Review by sethdarragile6 — APPROVED** [review]
LGTM!

**Pull request merged**